Cue/app/common/SyncConflict.js

```diff
@@ -0,0 +1,169 @@
+// @flow
+
+import React from 'react'
+import { View, Text, ListView, Dimensions, Navigator, Platform, Alert} from 'react-native'
+import { Deck } from '../api/types';
+import { connect } from 'react-redux'
+import CueIcons from './CueIcons'
+import CueHeader from './CueHeader'
+import { resolveConflict } from '../actions/library'
+import { SegmentedControls } from 'react-native-radio-buttons'
+
+  //TODO: style this page
+
+type Props = {
+  failedSyncs: Array<Deck>,
+  navigator: Navigator,
+
+  // Redux
+  localDecks: Array<Deck>
+}
+
+const styles = {
+  container: {
+    flex: 1,
+    backgroundColor: 'white',
+  },
+}
+
+class SyncConflict extends React.Component {
+  props: Props
+
+  state: {
+    dataSource: ListView.dataSource,
+    deviceOrientation: string,
+    conflicts: []
+  }
+
+  _onLayout = () => {
+    const windowDimensions = Dimensions.get('window')
+    this.setState({
+      ...this.state,
+      deviceOrientation: windowDimensions.width > windowDimensions.height ? 'LANDSCAPE' : 'PORTRAIT'
+    })
+  }
+
+  constructor(props: Props) {
+    super(props)
+    let ds = new ListView.DataSource({
+      rowHasChanged: (r1, r2) => r1 != r2
+    })
+    let conflicts = props.failedSyncs.map(change => {
+      return {
+        localDeck: props.localDecks.find(deck=> deck.uuid==change.uuid),
+        choice: undefined,
+      }
+    })
+    this.state = {
+      deviceOrientation: 'UNKNOWN',
+      dataSource: ds.cloneWithRows(conflicts),
+      conflicts,
+    }
+  }
+
+
+  _getLeftItem = () => {
+    return {
+      title: 'Cancel',
+      icon: CueIcons.cancel,
+      onPress: () => { this.props.navigator.pop() }
+    }
+  }
+
+  _getRightItems = () => {
+    if (this.state.conflicts && !this.state.conflicts.find(c => c.useServerDeck==undefined)) {
+      return [{
+        title: 'Done',
+        icon: CueIcons.done,
+        onPress: () => {
+          let promises = []
+          this.state.conflicts.forEach(conflict => {
+            promises.push(this.props.resolveConflict(conflict.useServerDeck, conflict.localDeck))
+          })
+          Promise.all(promises)
+          .then(this.props.navigator.pop())
+          .catch(e => {
+            console.error(e)
+            Alert.alert ('Alert', 'Failed to resolve conflicts')
+            this.props.navigator.pop()
+          })
+        }
+      }]
+    }
+  }
+
+  _setSelection = (deck: {}, useServerDeck: boolean) =>{
+    this.setState({
+      ...this.state,
```

Comment:
So actually I learned this recently, you don't need to unpack the current state when calling `setState` since React will automatically perform a shallow merge of the previous state.